test/test.js

```diff
@@ -271,6 +271,19 @@ describe('Puppeteer', function() {
       const args = puppeteer.defaultArgs();
       expect(args).toContain('--no-first-run');
     });
+    it('should dump browser process stderr', async({server}) => {
+      const {spawn} = require('child_process');
+      const path = require('path');
+      const options = Object.assign({dumpio: true}, defaultBrowserOptions);
+      const res = spawn('node',
+          [path.resolve('test', 'fixtures', 'dumpio.js'), PROJECT_ROOT, JSON.stringify(options), server.EMPTY_PAGE]);
+      const response = await new Promise(resolve => res.stderr.on('data', data => {
+        if (data.toString('utf8').includes('test'))
+          resolve(data.toString('utf8'));
+      }));
+
+      expect(response).toContain('test');
```

Comment:
you should await for the node script to terminate

Reply:
Done